test/extended/gssapi.sh

```diff
@@ -0,0 +1,205 @@
+export JUNIT_GSSAPI_OUTPUT="${LOG_DIR}/raw_test_output_gssapi.log"
+
+os::test::junit::declare_suite_start "${TEST_NAME}"
+
+os::cmd::expect_success 'hack/build-go.sh cmd/oc -tags=gssapi'
+os::cmd::expect_success_and_text 'oc version' 'GSSAPI Kerberos SPNEGO'
+
+function cleanup() {
+    out=$?
+    set +e
+    cleanup_openshift
+
+    # TODO(skuznets): un-hack this nonsense once traps are in a better state
+    if [[ -n "${JUNIT_REPORT_OUTPUT:-}" ]]; then
+      # get the jUnit output file into a workable state in case we crashed in the middle of testing something
+      os::test::junit::reconcile_output
+
+      # check that we didn't mangle jUnit output
+      os::test::junit::check_test_counters
+
+      # use the junitreport tool to generate us a report
+      "${OS_ROOT}/hack/build-go.sh" tools/junitreport
+      junitreport="$(os::build::find-binary junitreport)"
+
+      if [[ -z "${junitreport}" ]]; then
+          echo "It looks as if you don't have a compiled junitreport binary"
+          echo
+          echo "If you are running from a clone of the git repo, please run"
+          echo "'./hack/build-go.sh tools/junitreport'."
+          exit 1
+      fi
+
+      cat "${JUNIT_REPORT_OUTPUT}" "${JUNIT_GSSAPI_OUTPUT}"    \
+        | "${junitreport}" --type oscmd                        \
+                           --suites nested                     \
+                           --roots github.com/openshift/origin \
+                           --output "${ARTIFACT_DIR}/report.xml"
+      cat "${ARTIFACT_DIR}/report.xml" | "${junitreport}" summarize
+    fi
+
+    ENDTIME=$(date +%s); echo "$0 took $(($ENDTIME - $STARTTIME)) seconds"
+    exit $out
+}
+
+function update_auth_proxy_config {
+    local server_config="${1}"
+    local spec='{.items[0].spec.containers[0].env[?(@.name=="SERVER")].value}'
+    spec+='_'
+    spec+='{.items[0].status.conditions[?(@.type=="Ready")].status}'
+
+    os::cmd::expect_success "oc set env dc/gssapiproxy-server SERVER='${server_config}'"
+    os::cmd::try_until_text "oc get pods -l deploymentconfig=gssapiproxy-server -o jsonpath='${spec}'" "^${server_config}_True$"
+}
+
+function run_gssapi_tests() {
+    local image_name="${1}"
+    local server_config="${2}"
+    local container_exit_code_jsonpath='{.status.containerStatuses[0].state.terminated.exitCode}'
+    local oc_cmd_pod_exit_code="oc get pod '${image_name}' -o jsonpath='${container_exit_code_jsonpath}'"
+    local cmd_cat_pod_log="cat '${LOG_DIR}/${image_name}-${server_config}.log'"
+    oc run "${image_name}" \
+        --image="${PROJECT}/${image_name}" \
+        --generator=run-pod/v1 --restart=Never --attach \
+        --env=SERVER="${server_config}" \
+        1> "${LOG_DIR}/${image_name}-${server_config}.log" \
+        2>> "${JUNIT_GSSAPI_OUTPUT}"
+    # Lots of checks to really make sure that the tests ran successfully
+    os::cmd::expect_success_and_text "${cmd_cat_pod_log}" 'SUCCESS'
```

Comment:
I felt that having a long variable name used repeatedly was easier to understand than the whole command over and over.